File: _src/03/Chapter3.VB/Context.vb
Imports System.Data.Entity

Public Class Context
    Inherits DbContext
    Public Sub New()
        MyBase.New("name=chapter3")
    End Sub
    Property People As DbSet(Of Person)
    Property Phones() As DbSet(Of Phone)

    Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
        modelBuilder.Configurations.Add(New PersonMap)
    End Sub

    'Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
    '    modelBuilder.Entity(of Person)().Property(Function(p) p.FirstName) _ 
    '        .HasMaxLength(30)

    '        modelBuilder.Entity(of Person)().Property(Function(p) p.LastName) _
    '            .HasMaxLength(30)

    '        modelBuilder.Entity(of Person)().Property(Function(p) p.MiddleName) _
    '            .HasMaxLength(1).IsFixedLength().IsUnicode(False)
    'End Sub
End Class

File: _src/03/Chapter3.CSharp/Context.cs
using System.Data.Entity;

namespace Chapter3.CSharp
{
    public class Context : DbContext
    {
        public Context()
            : base("name=chapter3")
        {

        }
        public DbSet<Person> People { get; set; }
        public DbSet<Phone> Phones { get; set; }
protected override void OnModelCreating(DbModelBuilder modelBuilder)
{
    modelBuilder.Configurations.Add(new PersonMap());
}

        //protected override void OnModelCreating(DbModelBuilder modelBuilder)
        //{
        //    modelBuilder.Entity<Person>().Property(p => p.FirstName)
        //        .HasMaxLength(30);
        //    modelBuilder.Entity<Person>().Property(p => p.LastName)
        //        .HasMaxLength(30);
        //    modelBuilder.Entity<Person>().Property(p => p.MiddleName)
        //        .HasMaxLength(1)
        //        .IsFixedLength()
        //        .IsUnicode(false);
        //}
    }
}
